pkg/controller/direct/secretmanager/secret_controller.go

```diff
@@ -246,7 +258,10 @@ func (a *Adapter) Update(ctx context.Context, op *directbase.UpdateOperation) er
 	if err != nil {
 		return err
 	}
-
+	if paths.Has("ttl") {
+		paths = paths.Delete("ttl")
+		resource.Expiration = a.actual.Expiration
```

Comment:
Nit: there should be a comment here, because I don't think this does anything unless `expiration` is set in `paths`